gtda/diagrams/distance.py

```diff
@@ -221,4 +221,4 @@ def transform(self, X, y=None):
         if self.order is not None:
             Xt = np.linalg.norm(Xt, axis=2, ord=self.order)
 
-        return Xt
+        return np.swapaxes(Xt, 1, 0)
```

Comment:
We can easily avoid this costly operation just by switching the order of arguments on L#217.

```
        Xt = _parallel_pairwise(self._X, X2, self.metric,
```

to 


```
        Xt = _parallel_pairwise(X2, self._X, self.metric,
```

renaming X2 to X1 would also make more sense as a result.

Reply:
Thanks, that's clever, simple and better! I incorporated your suggestion.
I went for the `swapaxes` because it [seems to return a view](https://numpy.org/doc/1.18/reference/generated/numpy.swapaxes.html), but i haven't tested whether this causes any loss of performance.